File: TodoApi/Models/Docks/Domain/Dock.cs
using FrameworkDDD.Common;
using TodoApi.Models.Vessels;

namespace TodoApi.Models.Docks
{
    public class Dock : IAggregateRoot
    {
        public long Id { get; set; }

        // Human friendly identifier (name or number)
        public string Name { get; set; } = string.Empty;

        // Logical/location description inside the port (e.g., "North Pier 3")
        public string Location { get; set; } = string.Empty;

        // Physical characteristics
        public double Length { get; set; }
        public double Depth { get; set; }
        public double MaxDraft { get; set; }

        // Allowed vessel types that can berth at this dock (many-to-many)
        public ICollection<VesselType> AllowedVesselTypes { get; set; } = new List<VesselType>();
    }
}

File: frontend/src/app/models/dock.ts
export interface VesselTypeDTO {
  id: number;
  name: string;
}

export interface DockDTO {
  id: number;
  name: string;
  location: string;
  length: number;
  depth: number;
  maxDraft: number;
  allowedVesselTypes?: VesselTypeDTO[];
}

export interface CreateDockDTO {
  name: string;
  location: string;
  length: number;
  depth: number;
  maxDraft: number;
  allowedVesselTypeIds?: number[];
}

export interface UpdateDockDTO extends CreateDockDTO {
  id: number;
}
